### PR TITLE
SiteNotice: add missing translate() call and improve moment.js date comparison

### PR DESCRIPTION
**Pull request by jsnajdr**
I did two drive-by fixes in `SiteNotice` that is the only consumer of `getActiveDiscount`:
- add a missing `translate()` around an "Upgrade" label
- improve a moment.js "are both times the same day" comparison: there is a convenient API for that

None of these two fixes can be currently tested in real life, because no discounts define the `nudgeText` fields.

Found when working on the `lib/abtest` removal.

**How to test:**
Verify that unit tests pass.

### CODE REVIEW

**Review by tyxla — COMMENTED** [review]
@jsnajdr I'm already handling this removal in https://github.com/Automattic/wp-calypso/pull/53300/commits/53342359c0eaa232ad4e51d12e29d7c92ba8a743 and part of #53300. Is there any good reason to open a new PR for that part?

**Comment by jsnajdr** [comment]
> Is there any good reason to open a new PR for that part?

There is not, I'm just silly 🙂 I'll keep this PR open and then will rebase and repurpose it to make only the `translate` and `moment.isSame` fixes.

**Comment by tyxla** [comment]
Sounds good, happy to take a look when we rebase it after #53300 lands.

**Comment by jsnajdr** [comment]
@tyxla I just rebased and renamed this. Only two little fixes remained 🙂

File: client/my-sites/current-site/notice.jsx
Comment (tyxla) [inline client/my-sites/current-site/notice.jsx:244]: 😅

File: client/my-sites/current-site/notice.jsx
Comment (tyxla) [inline client/my-sites/current-site/notice.jsx:252]: Since `endsAt` is a simple `Date` object, we could easily avoid using `moment` by comparing just the date portion of the two date objects. We could do something like:

```suggestion
		return endsAt.toDateString() === new Date().toDateString();
```

**Review by tyxla — APPROVED** [review]
LGTM 👍  Just left one optional question/suggestion.

File: client/my-sites/current-site/notice.jsx
Comment (jsnajdr) [reply client/my-sites/current-site/notice.jsx:252]: As this way of comparing dates is not terribly elegant, I have another idea: what if we used this opportunity to introduce our first usage of `date-fns`? That would look like:
```js
import { isSameDay } from 'date-fns';

const same = isSameDay( endsAt, new Date() );
```

Otherwise, I'd rather keep the `moment` usage so that it can be easily found and replaced with `date-fns` in the future.

Comment (tyxla) [reply client/my-sites/current-site/notice.jsx:252]: I agree it's not that elegant, but it's native JS, it's clear and it works 😉  The options according to my personal preference are:

* `Date.prototype.toDateString()` - because it removes a usage of `moment` without introducing another library
* `moment` - because it leaves the code as it was from the perspective of library usage
* `date-fns` - because it introduces a new library (even if it's considered a better option than `moment`)

I'll leave it to your judgment as to what to go with, since I don't have a strong opinion on it and don't want to block the PR.

Comment (jsnajdr) [reply client/my-sites/current-site/notice.jsx:252]: I think that having to use `.toDateString()` shows gaps in the native APIs that will hopefully be addressed at some time. I'll stay with `moment` here so that future upgrade can be easily grepped. Thanks for your feedback :+1:

**Pull request merged**
